src/main/webapp/WEB-INF/view/notifications.jsp

```diff
@@ -25,19 +28,76 @@
+  <% int numUnreadNotifications = (int) request.getSession().getAttribute("numUnreadNotifications"); %>
+  <% int numReadNotifications = (int) request.getSession().getAttribute("numReadNotifications"); %>
+
   <div id="container">
 
     <% if(request.getAttribute("error") != null){ %>
         <h2 style="color:red"><%= request.getAttribute("error") %></h2>
     <% } %>
 
     <% if(request.getSession().getAttribute("user") != null){ %>
-    <h1>Hi <%= request.getSession().getAttribute("user") %>, you have _ notifications.</h1>
+      <% if(numUnreadNotifications==0){ %>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        no new notifications.</h1>
+      <% } else if(numUnreadNotifications==1){%>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        1 new notification.</h1>
+      <% } else{ %>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        <%= numUnreadNotifications %> new notifications.</h1>
+      <% } %>
+
+      <% if(numUnreadNotifications != 0){ %>
+      <h2>New Notifications</h2>
+      <div id="chat">
+        <ul>
+      <%
+        for (Notification notification : unreadNotifications) {
+          Message message = (Message) notification.getMessage();
+          String messageAuthor = message.getUser();
+          String conversationName = message.getConversation();
+          String messageConversationLink = "/chat/" + conversationName;
+          NotificationStore.getInstance().markNotificationAsViewed(notification);
+      %>
+        <li><strong><%= messageAuthor %></strong> mentioned you in <a href=<%= messageConversationLink %>><%= conversationName %></a>: "<%= message.getContent() %>"</li>
+      <%
+        }
+      %>
+        </ul>
+      </div>
+
+      <% } %>
 
+      <% if(numReadNotifications != 0){ %>
+      <h2>Read Notifications</h2>
+      <div id="chat">
+        <ul>
+      <%
+        for (Notification notification : readNotifications) {
+          Message message = (Message) notification.getMessage();
+          String messageAuthor = message.getUser();
+          String conversationName = message.getConversation();
+          String messageConversationLink = "/chat/" + conversationName;
+      %>
+        <li><strong><%= messageAuthor %></strong> mentioned you in <a href=<%= messageConversationLink %>><%= conversationName %></a>: "<%= message.getContent() %>"</li>
+      <%
+        }
+      %>
+        </ul>
+      </div>
 
-    <h2>Statistics</h2>
+      <% } %>
 
-    <p> Statistics here: 123 </p>
+      <% if(numReadNotifications > 0 || numUnreadNotifications > 0) { %>
+      <form action="/notifications" method="POST">
+        <button type="submit">Clear All</button>
+      </form>
+      <% } %>
+      
     <% } else{%>
         <h2 style="color:red">Error: Please log in to access notifications.</h2>
     <%}%>
```

Comment:
Nice! I noticed this when I was testing your app. A suggestion I have is to maybe add a link to /login with the "log in" text, but that's just minor